### PR TITLE
feat(components): [notification] add support for message as a Function

### PR DESCRIPTION
**Pull request by DDDDD12138**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### Description
This PR introduces support for the `message` prop in the `Notification` component to accept a `Function` type. This enhancement allows users to provide a function that returns a `VNode`.

image

[Preview Demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtYnV0dG9uIHBsYWluIEBjbGljaz1cIm9wZW4xXCI+IHN0cmluZyA8L2VsLWJ1dHRvbj5cbiAgPGVsLWJ1dHRvbiBwbGFpbiBAY2xpY2s9XCJvcGVuMlwiPiBWTm9kZSA8L2VsLWJ1dHRvbj5cbiAgPGVsLWJ1dHRvbiBwbGFpbiBAY2xpY2s9XCJvcGVuM1wiPiBmdW5jdGlvbiByZXR1cm4gVk5vZGUgPC9lbC1idXR0b24+XG48L3RlbXBsYXRlPlxuXG48c2NyaXB0IGxhbmc9XCJ0c1wiIHNldHVwPlxuaW1wb3J0IHsgaCB9IGZyb20gJ3Z1ZSdcbmltcG9ydCB7IEVsTm90aWZpY2F0aW9uIH0gZnJvbSAnZWxlbWVudC1wbHVzJ1xuXG5jb25zdCBvcGVuMSA9ICgpID0+IHtcbiAgRWxOb3RpZmljYXRpb24oe1xuICAgIHRpdGxlOiAnVGl0bGUnLFxuICAgIG1lc3NhZ2U6ICdzdHJpbmcnLFxuICB9KVxufVxuXG5jb25zdCBvcGVuMiA9ICgpID0+IHtcbiAgRWxOb3RpZmljYXRpb24oe1xuICAgIHRpdGxlOiAnUHJvbXB0JyxcbiAgICBtZXNzYWdlOiBoKCdkaXYnLCBudWxsLCAnVm5vZGUgcmVuZGUnKSxcbiAgfSlcbn1cblxuY29uc3Qgb3BlbjMgPSAoKSA9PiB7XG4gIEVsTm90aWZpY2F0aW9uKHtcbiAgICB0aXRsZTogJ1Byb21wdCcsXG4gICAgbWVzc2FnZTogKCkgPT4gaCgnZGl2JywgbnVsbCwgJ3JldHVybiBWbm9kZSByZW5kZScpLFxuICB9KVxufVxuPC9zY3JpcHQ+IiwiZWxlbWVudC1wbHVzLmpzIjoiaW1wb3J0IEVsZW1lbnRQbHVzIGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IGdldEN1cnJlbnRJbnN0YW5jZSB9IGZyb20gJ3Z1ZSdcblxubGV0IGluc3RhbGxlZCA9IGZhbHNlXG5hd2FpdCBsb2FkU3R5bGUoKVxuXG5leHBvcnQgZnVuY3Rpb24gc2V0dXBFbGVtZW50UGx1cygpIHtcbiAgaWYgKGluc3RhbGxlZCkgcmV0dXJuXG4gIGNvbnN0IGluc3RhbmNlID0gZ2V0Q3VycmVudEluc3RhbmNlKClcbiAgaW5zdGFuY2UuYXBwQ29udGV4dC5hcHAudXNlKEVsZW1lbnRQbHVzKVxuICBpbnN0YWxsZWQgPSB0cnVlXG59XG5cbmV4cG9ydCBmdW5jdGlvbiBsb2FkU3R5bGUoKSB7XG4gIGNvbnN0IHN0eWxlcyA9IFsnaHR0cHM6Ly9wcmV2aWV3LTE4NTU4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0xODU1OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2luZGV4LmNzcyddLm1hcCgoc3R5bGUpID0+IHtcbiAgICByZXR1cm4gbmV3IFByb21pc2UoKHJlc29sdmUsIHJlamVjdCkgPT4ge1xuICAgICAgY29uc3QgbGluayA9IGRvY3VtZW50LmNyZWF0ZUVsZW1lbnQoJ2xpbmsnKVxuICAgICAgbGluay5yZWwgPSAnc3R5bGVzaGVldCdcbiAgICAgIGxpbmsuaHJlZiA9IHN0eWxlXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2xvYWQnLCByZXNvbHZlKVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdlcnJvcicsIHJlamVjdClcbiAgICAgIGRvY3VtZW50LmJvZHkuYXBwZW5kKGxpbmspXG4gICAgfSlcbiAgfSlcbiAgcmV0dXJuIFByb21pc2UuYWxsU2V0dGxlZChzdHlsZXMpXG59XG4iLCJ0c2NvbmZpZy5qc29uIjoie1xuICBcImNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogXCJFU05leHRcIixcbiAgICBcImpzeFwiOiBcInByZXNlcnZlXCIsXG4gICAgXCJtb2R1bGVcIjogXCJFU05leHRcIixcbiAgICBcIm1vZHVsZVJlc29sdXRpb25cIjogXCJCdW5kbGVyXCIsXG4gICAgXCJ0eXBlc1wiOiBbXCJlbGVtZW50LXBsdXMvZ2xvYmFsLmQudHNcIl0sXG4gICAgXCJhbGxvd0ltcG9ydGluZ1RzRXh0ZW5zaW9uc1wiOiB0cnVlLFxuICAgIFwiYWxsb3dKc1wiOiB0cnVlLFxuICAgIFwiY2hlY2tKc1wiOiB0cnVlXG4gIH0sXG4gIFwidnVlQ29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiAzLjNcbiAgfVxufVxuIiwiUGxheWdyb3VuZE1haW4udnVlIjoiPHNjcmlwdCBzZXR1cD5cbmltcG9ydCBBcHAgZnJvbSAnLi9BcHAudnVlJ1xuaW1wb3J0IHsgc2V0dXBFbGVtZW50UGx1cyB9IGZyb20gJy4vZWxlbWVudC1wbHVzLmpzJ1xuc2V0dXBFbGVtZW50UGx1cygpXG48L3NjcmlwdD5cblxuPHRlbXBsYXRlPlxuICA8QXBwIC8+XG48L3RlbXBsYXRlPlxuIiwiaW1wb3J0LW1hcC5qc29uIjoie1xuICBcImltcG9ydHNcIjoge1xuICAgIFwidnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3J1bnRpbWUtZG9tQGxhdGVzdC9kaXN0L3J1bnRpbWUtZG9tLmVzbS1icm93c2VyLmpzXCIsXG4gICAgXCJAdnVlL3NoYXJlZFwiOiBcImh0dHBzOi8vZmFzdGx5LmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9zaGFyZWRAbGF0ZXN0L2Rpc3Qvc2hhcmVkLmVzbS1idW5kbGVyLmpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXNcIjogXCJodHRwczovL3ByZXZpZXctMTg1NTgtZWxlbWVudC1wbHVzLnN1cmdlLnNoL2J1bmRsZS9pbmRleC5mdWxsLm1pbi5tanNcIixcbiAgICBcImVsZW1lbnQtcGx1cy9cIjogXCJ1bnN1cHBvcnRlZFwiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7InNob3dIaWRkZW4iOnRydWUsInN0eWxlU291cmNlIjoiaHR0cHM6Ly9wcmV2aWV3LTE4NTU4LWVsZW1lbnQtcGx1cy5zdXJnZS5zaC9idW5kbGUvaW5kZXguY3NzIn19)

### CODE REVIEW

File: docs/en-US/component/notification.md [hunk 1]
```diff
@@ -111,7 +111,7 @@ ElNotification({}, appContext)
 | Name                     | Description                                                                                                        | Type                                                                  | Default   |
 | ------------------------ | ------------------------------------------------------------------------------------------------------------------ | --------------------------------------------------------------------- | --------- |
 | title                    | title                                                                                                              | ^[string]                                                             | ''        |
-| message                  | description text                                                                                                   | ^[string] / ^[VNode]                                                  | ''        |
+| message                  | description text                                                                                                   | ^[string] / ^[VNode] / ^[Function]`() => VNode`                       | ''        |
```
Comment (btea) [inline docs/en-US/component/notification.md:114]: Perhaps we can add an example to the document, and then add a version tag after the example to explain the version that supports configuring the parameter value as a function.

Comment (DDDDD12138) [reply docs/en-US/component/notification.md:126]: image

I added this example, is that okay?

File: docs/en-US/component/notification.md [hunk 2]
```diff
@@ -61,6 +61,18 @@ Although `message` property supports HTML strings, dynamically rendering arbitra
 
 :::
 
+## Use VNode
```
Comment (btea) [inline docs/en-US/component/notification.md:64]:
```suggestion
## Message using functions ^(2.8.7)
```
Put the version number in the title to make it more visible.

**Review by btea — APPROVED** [review]
Thanks!

File: docs/en-US/component/notification.md (hunk 2 above)
Comment (warmthsea) [reply docs/en-US/component/notification.md]: > Put the version number in the title to make it more visible.

The documentation here seems to have some issues, this change seems unnecessary, what do you think?

1. `Vnode` was supported from the beginning, but it was not documented

image


2. This time, `function` support was added (`2.8.7`), and the return type is specifically `Vnode`, which conflicts with the title

image

**Comment by DDDDD12138** [comment]
Thank you for your review. @btea @warmthsea 

How should I proceed with the final revisions? Please give me some suggestions.

File: docs/en-US/component/notification.md (hunk 2 above)
Comment (btea) [reply docs/en-US/component/notification.md]: > 1. `Vnode` was supported from the beginning, but it was not documented

`message` supports `vnode` type, and the document also marks this type. What is the problem you are referring to?

> 2. This time, `function` support was added (`2.8.7`), and the return type is specifically `Vnode`, which conflicts with the title

What exactly is the conflict you are talking about?

Comment (warmthsea) [reply docs/en-US/component/notification.md]: I missed your reply, I review again, no problem.

**Comment by warmthsea** [comment]
> Thank you for your review. @btea @warmthsea
> 
> How should I proceed with the final revisions? Please give me some suggestions.

@DDDDD12138 
Could you please change the version number to 2.9.0 ?

**Review by warmthsea — APPROVED** [review]
Thanks!

**Pull request merged**